pkg/controllers/uiplugin/troubleshooting_panel.go

```diff
@@ -99,3 +91,29 @@ func marshalTroubleshootingPanelPluginConfig(cfg *uiv1alpha1.TroubleshootingPane
 
 	return buf.String(), nil
 }
+
+func getLokiServiceName(ctx context.Context, k client.Client, ns string) (string, error) {
+
+	serviceList := &corev1.ServiceList{}
+	if err := k.List(ctx, serviceList, client.InNamespace(ns)); err != nil {
+		return "", err
+	}
+
+	// Accumulate services that contain "gateway" in their names
+	var gatewayServices []corev1.Service
+	for _, service := range serviceList.Items {
+		if strings.Contains(service.Name, "gateway") {
+			gatewayServices = append(gatewayServices, service)
+		}
+	}
+	key := "app.kubernetes.io/managed-by"
+	if len(gatewayServices) > 0 {
```

Comment:
This check is not necessary

Reply:
So just return the first service from the list, since loki is the only gateway service expected? Is that what you meant?

Reply:
Ohh sorry I was wrong. I thought there was a for loop so apparently I was looking the wrong way. You can ignore. Sorry.